src/main/java/jenkins/plugins/maveninfo/extractor/properties/PomPropertiesFinder.java

```diff
@@ -51,32 +51,33 @@ public void findProperties(ExtractorContext ctx) throws IOException,
 				.getCompiledMainModulePattern();
 
 		MavenModule main = BuildUtils.getMainModule(mmsb, mainPattern);
-		if (main != null) {
-			
-			MavenBuild build = mmsb.getModuleLastBuilds().get(main);
-			if (build != null) {
-				
-				FilePath p = build.getWorkspace().child("pom.xml");
-				if (p != null) {
-
-					InputStream is = null;
-					try {
-						Digester digester = createDigester(!Boolean.getBoolean(this.getClass().getName() + ".UNSAFE"));
-						digester.setRules(new ExtendedBaseRules());
-
-						ctx.getRuleSet().addRuleInstances(digester);
-
-						is = p.read();
-						digester.parse(is);
-					} catch (SAXException ex) {
-						throw new IOException("Can't read POM: " + p.toString());
-					} finally {
-						if (is != null) {
-							is.close();
-						}
-					}
-				}
+		if (main == null) {
+			return;
+		}
+
+		MavenBuild build = mmsb.getModuleLastBuilds().get(main);
+		if (build == null) {
+			return;
+		}
+
+		FilePath workspace = build.getWorkspace();
+		if (workspace == null) {
+			return;
+		}
+
+		FilePath p = workspace.child("pom.xml");
+
+		try {
+			Digester digester = createDigester(!Boolean.getBoolean(this.getClass().getName() + ".UNSAFE"));
+			digester.setRules(new ExtendedBaseRules());
+
+			ctx.getRuleSet().addRuleInstances(digester);
+
+			try (InputStream is = p.read()) {
```

Comment:
good but maybe having only one single `try` here it makes the code a bit confusing.
sorry for the nit

Reply:
No worries and agreed 😄